src/Catalog/Helpers/ParallelExecutionUtility.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    public static class ParallelExecutionUtility
+    {
+        public const int DefaultDegreeOfParallelism = 32;
+
+        /// <summary>
+        /// Runs a task created by <paramref name="taskFactory"/> on a number of threads specified by <paramref name="degreeOfParallelism"/> and waits for all tasks to complete.
+        /// </summary>
+        /// <param name="taskFactory">Creates tasks for each thread to run.</param>
+        /// <param name="degreeOfParallelism">The number of threads to run tasks on.</param>
+        /// <returns>A task that completes when all threads have completed their tasks.</returns>
+        public static Task ProcessInParallel(Func<Task> taskFactory, int degreeOfParallelism = DefaultDegreeOfParallelism)
```

Comment:
Need a way to cancel the operation.

Reply:
Deleting class so no longer necessary

Reply:
The user can cancel the tasks they provide to this, which should provide the same functionality, but I will add a cancellation token

Reply:
`Task.WhenAll` doesn't support passing a `CancellationToken`, so all `CancellationToken`s must be set up by the user in the `Func<Task> taskFactory`.